### ISSUE TITLE
Errors in package testing don't cause tests to be marked as failed

### ISSUE THREAD

**Issue by queso** [Sanjo/meteor-jasmine#257]
I am testing publications in a jasmine package test and an exception from the publication doesn't cause the test to fail...

```
I20150728-12:54:11.297(-4)? Exception in callback of async function: Error: Match error: Expected string, got undefined
I20150728-12:54:11.297(-4)?   at checkSubtree (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/check.js:177:13)
I20150728-12:54:11.297(-4)?   at check (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/check.js:50:5)
I20150728-12:54:11.297(-4)?   at Object.Meteor.publish.Orders.find.shopId (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/reactioncommerce_core.js:3468:3)
I20150728-12:54:11.297(-4)?   at /private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/matb33_collection-hooks.js:242:21
I20150728-12:54:11.297(-4)?   at [object Object]._.extend.withValue (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/meteor.js:989:17)
I20150728-12:54:11.298(-4)?   at Object.Product (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/matb33_collection-hooks.js:241:28)
I20150728-12:54:11.298(-4)?   at /private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/local-test_reactioncommerce_core.js:1166:59
I20150728-12:54:11.298(-4)?   at runWithEnvironment (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/meteor.js:1041:24)
I20150728-12:54:11.298(-4)?   at Object.<anonymous> (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/meteor.js:1054:14)
I20150728-12:54:11.298(-4)?   at Object.<anonymous> (/private/var/folders/4z/ymnx83bx17q8lf38qhhm_gnh0000gn/T/meteor-test-run5o6yyl/.meteor/local/build/programs/server/packages/sanjo_jasmine.js:292:32)
I20150728-12:54:11.298(-4)?   at attemptSync (/Users/josh/.meteor/packages/sanjo_jasmine/.0.15.1.1bof1hq++os+web.browser+web.cordova/npm/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1759:24)
I20150728-12:54:11.298(-4)?   at QueueRunner.run (/Users/josh/.meteor/packages/sanjo_jasmine/.0.15.1.1bof1hq++os+web.browser+web.cordova/npm/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1747:9)
```

The call is just `cursor = Meteor.server.publish_handlers["Product"]();`

**Comment by Sanjo** [comment]
Is the test publicly available?

**Comment by queso** [comment]
If you remove the product._id from this line of code and just make an empty call, it will trigger the issue:

https://github.com/reactioncommerce/reaction-core/pull/154/files#diff-0c8cde99bb897255fe42eb5007e41106R63

**Comment by queso** [comment]
Also, if I pass an undefined var to expect(data) and make a call on it, like expect(data.title), it will trigger an error, but the test still passes.

**Comment by Sanjo** [comment]
Confirmed. I will fix it this week.

**Comment by Sanjo** [comment]
Fixed with 0.15.2. It was a regression of 0.15.1. https://github.com/Sanjo/meteor-jasmine/commit/fd2f686c12a2393e057ebe47cac12fff14cf3feb

**Comment by Sanjo** [comment]
Duplicate of https://github.com/Sanjo/meteor-jasmine/issues/229
